parser/src/lib.rs

```diff
@@ -72,6 +75,22 @@ pub fn parse_module<'a, T: FieldElement>(
         .map_err(|err| handle_parse_error(err, file_name, input))
 }
 
+pub fn parse_type_name<T: FieldElement>(
+    input: &str,
+) -> Result<powdr_ast::parsed::TypeName<powdr_ast::parsed::Expression<T>>, ParseError<'_>> {
+    let ctx = ParserContext::new(None, input);
+    TYPE_NAME_PARSER
+        .parse(&ctx, input)
+        .map_err(|err| handle_parse_error(err, None, input))
+}
+
+pub fn parse_type_var_bounds(input: &str) -> Result<TypeBounds, ParseError<'_>> {
+    let ctx = ParserContext::new(None, input);
+    TYPE_VAR_BOUNDS_PARSER
+        .parse::<GoldilocksField>(&ctx, input)
```

Comment:
why Goldilocks?

Reply:
True, this should be generic. We really need to change the grammar to use integers instead...

Reply:
But what happens with Bn then?